.github/workflows/deploy.yml

```diff
@@ -1,68 +1,205 @@
-
-#      - name: Build, tag, and push image to Amazon ECR
-#        id: build-image
-#        env:
-#          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
-#          ECR_REPOSITORY: ${{ env.ecr_repository }}
-#          IMAGE_TAG: ${{ github.sha }}
-#        run: |
-#          # Build a docker container and
-#          # push it to ECR so that it can
-#          # be deployed to ECS.
-#          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
-#          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
-
-#          # Retag this as the latest
-#          docker tag $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG $ECR_REGISTRY/$ECR_REPOSITORY:latest
-#          docker push $ECR_REGISTRY/$ECR_REPOSITORY:latest
-#          echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG"
-
-#      #- name: Fill in the new image ID in the Amazon ECS task definition
-#      #  id: task-def
-#      #  uses: aws-actions/amazon-ecs-render-task-definition@v1
-#      #  with:
-#      #    task-definition: ${{ env.task_definition }}
-#      #    container-name: ${{ env.container_name }}
-#      #    image: ${{ steps.build-image.outputs.image }}
-
-#      #- name: Deploy Amazon ECS task definition
-#      #  uses: aws-actions/amazon-ecs-deploy-task-definition@v1
-#      #  with:
-#      #    task-definition: ${{ steps.task-def.outputs.task-definition }}
-#      #    service: ${{ env.service_name }}
-#      #    cluster: ${{ env.cluster_name }}
-#      #    wait-for-service-stability: true
+name: Deploy to Amazon ECS
+
+env:
+  aws_region: eu-west-2
+  task_definition_name: webservers
+  service_name: webservers
+  cluster_name: webservers
+
+  ecr_repository_rails: "webserver-rails"
+  dockerfile_rails: "docker/rails.Dockerfile"
+
+  ecr_repository_nginx: "webserver-nginx"
+  dockerfile_nginx: "docker/nginx.Dockerfile"
+
+  ecr_repository_anycable_go: "webserver-anycable-go"
+  dockerfile_anycable_go: "docker/anycable-go.Dockerfile"
+
+  # TODO: Move this to be a secret
+  s3_bucket: "exercism-assets-staging"
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    name: Upload images to ECR, assets to S3, and defintion to ECS
+    runs-on: ubuntu-latest
+    steps:
+      ###############
+      ###############
+      # Setup Steps #
+      ###############
+      ###############
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: satackey/action-docker-layer-caching@v0.0.5
+        continue-on-error: true
+        with:
+          # Change these to rest teh cache
+          key: docker-layer-caching-deploy-{hash}
+          restore-keys: docker-layer-caching-deploy-
+          concurrency: 30
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_DEPLOY_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_DEPLOY_SECRET_ACCESS_KEY }}
```

Comment:
These are org-secrets that point to keys for a user called `github-deploy`. That user's permissions are defined in [these files](https://github.com/exercism/terraform/tree/master/terraform/github_deploy) in the terraform repo, are are effectively limited to reading and writing to the relevant ECR repositories, getting and pushing new task-definitions and spinning a new service up.